crates/iroha_schema/src/lib.rs

```diff
@@ -280,6 +280,22 @@ pub struct BitmapMask {
 #[derive(Debug, Clone, Serialize)]
 pub struct Compact<T>(T);
 
+impl TypeId for () {
+    fn id() -> String {
+        "()".to_owned()
+    }
+}
+impl IntoSchema for () {
+    fn type_name() -> String {
+        "()".to_owned()
+    }
+    fn update_schema_map(map: &mut MetaMap) {
+        if !map.contains_key::<Self>() {
+            map.insert::<Self>(Metadata::Tuple(UnnamedFieldsMeta { types: vec![] }));
+        }
+    }
+}
```

Comment:
why did you need this?

Reply:
Basically, because I share the projections between predicates and selectors.

As decided by impl of `Projectable` trait, predicates store actual atomic predicates, while query selectors store `()` because there's nothing else to store there.